### PR TITLE
Flip strict=True for //enterprise/app/root

### PR DESCRIPTION
**Pull request by iain-macdonald**
Version bump: Patch
Related issues: Unblocks https://github.com/buildbuddy-io/buildbuddy-internal/issues/936

### CODE REVIEW

File: app/terminal/terminal.tsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ export default class TerminalComponent extends React.Component<TerminalProps, St
     window.removeEventListener("keydown", this.windowKeyDownListener);
   }
 
-  componentDidUpdate(_prevProps: TerminalProps, prevState: State, snapshot: Snapshot | null) {
+  componentDidUpdate(_prevProps: TerminalProps, prevState: State, snapshot: Snapshot) {
```
Comment (bduffany) [inline app/terminal/terminal.tsx:100]: I think maybe this needs to be `snapshot?: Snapshot` based on the type signature here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/522a694de4e20d1ab69598affc0a97d2ee8d2556/types/react/v16/index.d.ts#L681

I remember hitting an issue that snapshot was either null or undefined which is why there's the `if (snapshot && ...)` check below

File: app/root/root.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ import UserPreferences from "../preferences/preferences";
 declare var window: any;
 
 interface State {
-  user: User;
+  user?: User;
```
Comment (bduffany) [inline app/root/root.tsx:19]: Seems like the tests are failing -- I wonder if we have some logic somewhere that is checking specifically for `user === null`. Might be less risky to change this to `user: User | null`

**Review by bduffany — APPROVED** [review]

File: app/terminal/terminal.tsx (hunk 1 above)
Comment (iain-macdonald) [reply app/terminal/terminal.tsx]: Ahh yeah good catch.

File: app/root/root.tsx (hunk 2 above)
Comment (iain-macdonald) [reply app/root/root.tsx]: Per discussion it's this code in invocation.tsx: https://github.com/buildbuddy-io/buildbuddy/blob/e753ffa527b2fb96b4095d4ee6acb29c32dfbc0f/app/invocation/invocation.tsx#L152

But I flipped it back to "| null" and added a todo to change it to optional in the future.

**Pull request merged**